File: python/day15.py
zoo = list(range(1,10))
for zo in zoo:
	if zo == 1:
		z = '1st'
	elif zo == 2:
		z = '2nd'
	elif zo == 3:
		z = '3rd'
	else:
		z = str(zo)+'th'
	print(z)

# 运算里的加减乘除
c = 328
d = 7
print(c+d)
print(c-d)
print(c*d)
print(c/d)
print(c//d) #整除运算
print(c**d) #幂运算
print(c%d) #求模(余)运算
""" 
整除和求模的应用场景
行和列位置的应用 
"""

#
a = int(input('a = '))
b = int(input('b = '))
print('%d + %d = %d' % (a, b, a + b))
print('%d - %d = %d' % (a, b, a - b))
print('%d * %d = %d' % (a, b, a * b))
print('%d / %d = %f' % (a, b, a / b))
print('%d // %d = %d' % (a, b, a // b))
print('%d %% %d = %d' % (a, b, a % b))
print('%d ** %d = %d' % (a, b, a ** b))

num = 10
num += 5
# num = num + 5
print(num)

# id是内存地址的标识符
name = 15
print(id(num))

# is是比较id标识符是否相等的比较运算
print(num is name)

# 非布尔的值，如果作为真假来判定，非零即真，非空即真
print(name and num)

File: C/day15.c
// 24.02.23
/*
// do while循环
// 至少先执行一次do语句,再进行条件判断后是否循环,总结：先干为敬
# include <stdio.h>
int main(){
    int g =0;
    do{
        printf("无限进步！🤩");
        g++;
        }
    while (g < 2);
    return 0;
}



// n的阶乘
# include <stdio.h>
int main(){
    int n = 0;
    int s = 1;
    int a = 1;
    printf("请输入一个整数：\n");
    scanf("%d", &n);
    for (s = 1; s <= n; s++)
        a = a * s;
        printf("%d! = %d",n,a);
    return 0;
}
*/
// 1!+2!+3!+...10!
# include <stdio.h>
int main(){
    int n = 1;
    int s = 1;
    int a = 1;
    int num = 0;
    for (n = 1; n <= 10;n++){ //注意括号范围
            a = a * n;
            num = a + num;
            }
    printf("%d",num);
    return 0;
}

